nikola/utils.py

```diff
@@ -1938,19 +1943,19 @@ def sort_posts(posts, *keys):
     return posts
 
 
-def smartjoin(join_char: str, string_or_iterable) -> str:
-    """Join string_or_iterable with join_char if it is iterable; otherwise converts it to string.
+def smartjoin(join_char: str, string_or_iterable: Union[None, str, bytes, Iterable[Any]]) -> str:
+    """Join string_or_iterable with join_char if it is iterable; otherwise convert it to string.
 
     >>> smartjoin('; ', 'foo, bar')
     'foo, bar'
     >>> smartjoin('; ', ['foo', 'bar'])
     'foo; bar'
     >>> smartjoin(' to ', ['count', 42])
     'count to 42'
+
+    The treatment of bytes (calling str(string_or_iterable)) is somewhat dubious. Is this needed?
```

Comment:
Might be worth checking if `bytes` is ever passed here.

Reply:
Yes!